formatter.go

```diff
@@ -472,6 +497,68 @@ func (f *DefaultFormatter) formatFloatValue(value float64, bits int) string {
 	}
 }
 
+func (f *DefaultFormatter) addDigitGrouping(numStr string) string {
+	var sign string
+	if numStr[0] == '-' || numStr[0] == '+' {
+		sign = string(numStr[0])
+		numStr = numStr[1:]
+	}
+
+	parts := strings.SplitN(numStr, ".", 2)
```

Comment:
To make this code easier to follow, I suggest to extract a function, say, parseNumber, that returns all parts of the number: integer, fractional, and exponent.

Here we can call it, and apply groupping to integer and fractional parts.

Then we can compose number from parts.